clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
Also if a consumer start with subscribe, then switch to manual assign, and then switch back to subscribe again, it will reuse its old consumer id in the join group. Is this intentional behavior?